mne_bids/path.py

```diff
@@ -215,6 +212,8 @@ class BIDSPath(object):
         The basename of the file path. Similar to `os.path.basename(fpath)`.
     root : str
         The root of the BIDS path.
+    directory : pathlib.Path
```

Comment:
why do we need this?

bids_path.fpath.parent ?

Reply:
So since fpath sometimes tries to infer the full filepath, then getting its parent is simply the directory `<root>/sub-<sub>/ses-<ses>/<datatype>`, but if the fpath errors out, then you can't get its parent. This might happen if you don't add enough entities, and it is unparseable. 

I don't think we've settled on the best way to handle that case, but for now, if we expose directory, then this would solve the issue? 

WDYT @hoechenberger

Reply:
Shouldn't we prevent the construction of a `BIDSPath` that lacks all the info we require to construct a sensible path in the first place?

The following
```
BIDSPath(root='/foo/', subject='01')
```
should probably error out on instantiation / update (subject supplied, but data type missing -> one directory level remains unspecified), unless one explicitly requests the operation via a special kwarg (`force=True` or something).

Reply:
Well, what if someone wants to create the subjects folder? Then it's sensible they would want to do:

```Python
bids_path = BIDSPath(root='/foo/', subject='01')
bids_path.mkdir()
subj_path = bids_path.directory

# errors out if you have multiple runs for subject '01'
bids_path.fpath()
```